src/components/SearchComponent/index.js

```diff
@@ -0,0 +1,96 @@
+import { useState } from "react";
+
+import "./styles.css";
+import useCity from "../CustomHooks/useCity";
+import StateCityInformation from "../../common/utils/state-city.json";
+
+const SearchComponent = ({ setCurrentCards, teamInformation }) => {
+  const [nameOrDesignation, setNameOrDesignation] = useState("");
+  const [state, setState] = useState("");
+  const [cityName, setCityName] = useState("");
+
+  const city = useCity(state);
+  const handleOnSubmit = (event) => {
+    event.preventDefault();
+    if (!cityName) {
+      setCurrentCards(
+        teamInformation.filter(
```

Comment:
Here you can make a different function to handle logic inside filter. Code readability is much less here.